### PR TITLE
docs: Update the documentation for the repo.

### PR DESCRIPTION
**Pull request by karthikbekalp**
### What was the problem/requirement? (What/Why)

Our documentation for this repo was not good enough for easy onboarding for customers. 

### What was the solution? (How)

Hence, updated the documentation accordingly. This PR is inspired mostly from a similar change in the Maya Github repo for deadline cloud : https://github.com/aws-deadline/deadline-cloud-for-maya/pull/173

This also adds submitter installation instructions for Mac OS. https://github.com/aws-deadline/deadline-cloud-for-cinema-4d/issues/61

### What is the impact of this change?

Better onboarding steps for customers. Mainly moved the contents of `DEVELOPMENT.md` to `README.md` as customers onboarding shouldn't have to deep dive into the `DEVELOPMENT.md` page. 

We will update `DEVELOPMENT` page in a follow up PR but for now we want to make the `README` page with clear instructions to follow. 

### How was this change tested?

This was based off of my personal notes that helped me onboard to Cinema 4D easily. 

### Was this change documented?

Yes

### Is this a breaking change?

No
----

*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,37 +13,147 @@ Our focus is to explore a variety of software applications to ensure we have goo
...
+
+```
+pip install deadline-cloud-for-cinema-4d
+pip install deadline[gui]
+```
+
+2. Set up the `C4DPYTHONPATH311` environment variable:
+  - Windows: 
+```
+set C4DPYTHONPATH311"Path/to/site-packages"
```
Comment (crowecawcaw) [inline README.md]: Missing a `=`

File: README.md [hunk 2]
```diff
@@ -13,37 +13,147 @@ Our focus is to explore a variety of software applications to ensure we have goo
...
+
+Windows:
+```
+set g_additionalModulePath="Path\to\DeadlineCloud.pyp"
+```
+
+Linux or Mac:
+```
+export g_additionalModulePath="Path\to\DeadlineCloud.pyp"
+```
```
Comment (crowecawcaw) [inline README.md]: The path should use forward slashes.

File: README.md [hunk 3]
```diff
@@ -13,37 +13,147 @@ Our focus is to explore a variety of software applications to ensure we have goo
...
 ## Compatibility
 
 This library requires:
 
-1. Cinema 4D 2023 - 2024,
-1. Python 3.9 or higher; and
-1. Linux, Windows, or a macOS operating system.
+1. Cinema 4D 2024 - 2025
+1. Python 3.9 or higher; but Python 3.11 is recommended as this is the version Cinema 4D uses natively.
+1. Windows
```
Comment (crowecawcaw) [inline README.md]: There are some instructions for Linux and Mac. Are they partially working but not supported? Might be helpful to clarify that here.

File: README.md [hunk 4]
```diff
@@ -13,37 +13,147 @@ Our focus is to explore a variety of software applications to ensure we have goo
...
+- Set the above environment variables in a shell. 
+- Navigate to the location where the Cinema 4D application is present.
+Normally it is in `/Applications/Maxon Cinema 4D 2025/Cinema 4D.app/Contents/MacOS`
+- Run `./Cinema\ 4D`
+
+If you load a scene, click on **Extensions > Deadline Cloud Submitter** to view the submitter.
+
+### Additional Python Libraries
+
+Some specific versions of Cinema 4d ( e.g. `Cinema 4D 2024.1.0`) have been found to be missing some libraries key to Deadline requirements ; in later versions such as `2024.4.0` this has been resolved.
```
Comment (crowecawcaw) [inline README.md]: Super nit: `4d` -> `4D`

**Review by crowecawcaw — CHANGES_REQUESTED** [review]

File: hatch.toml [hunk 5]
```diff
@@ -19,7 +19,6 @@ lint = [
   "style",
   "typing",
 ]
-install = "bash {root}/scripts/install_dev_submitter.sh {args:}"
```
Comment (karthikbekalp) [inline hatch.toml:22]: We do not have this script and hence I removed it from `hatch.toml`

File: README.md (hunk 3 above)
Comment (karthikbekalp) [reply README.md]: > Are they partially working but not supported?

We can support Mac submitter but Mac does not have an official commandline rendering application for Mac so we don't support it for the adaptors for Cinema 4D. 

Linux cannot be installed as submitters as only commandling rendering is available but it has not been fully tested. 

I'll update it here.

File: README.md (hunk 1 above)
Comment (karthikbekalp) [reply README.md]: Oops, will fix this right away.

File: README.md (hunk 2 above)
Comment (karthikbekalp) [reply README.md]: Oh makes sense, I'll update this right away.

File: README.md (hunk 4 above)
Comment (karthikbekalp) [reply README.md]: Makes sense, will update it right away.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -6,10 +12,47 @@
...
+See [DEVELOPMENT.md](https://github.com/aws-deadline/deadline-cloud-for-cinema-4d/blob/mainline/DEVELOPMENT.md) for information on running tests.
+
+- Have you run the unit tests?
+*delete text ending here*
+
+### Did you run the "Job Bundle Output Tests"? If not, why not? If so, paste the test results here.
+
+*delete text starting here*
+See the "Integration Tests" subsection of the
+[Running Submitter Tests](https://github.com/aws-deadline/deadline-cloud-for-cinema-4d/blob/mainline/DEVELOPMENT.md#running-submitter-tests)
```
Comment (joel-wong-aws) [inline .github/PULL_REQUEST_TEMPLATE.md]: Do we have this section for C4D? I don't see it

File: README.md [hunk 7]
```diff
@@ -13,37 +13,147 @@ Our focus is to explore a variety of software applications to ensure we have goo
...
 ## Compatibility
 
 This library requires:
 
-1. Cinema 4D 2023 - 2024,
-1. Python 3.9 or higher; and
-1. Linux, Windows, or a macOS operating system.
+1. Cinema 4D 2024 - 2025
+1. Python 3.9 or higher; but Python 3.11 is recommended as this is the version Cinema 4D uses natively.
+1. Windows is recommended; We have some information below on how to setup the submitter on Mac and adaptors on Linux but it is more experimental.
```
Comment (joel-wong-aws) [inline README.md]: nit: `more experimental` is ambiguous - either it is experimental or it isn't

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 6 above)
Comment (karthikbekalp) [reply .github/PULL_REQUEST_TEMPLATE.md]: Hmm, initially I wanted to also update the DEVELOPMENT.md but we have had issues with customers not able to onboard properly due to outdated README.md. So I wanted to get this in as quickly as possible. (I mentioned it in the PR description above). 

Anyway removed it from here and I'll add it once I've updated `DEVELOPMENT.md`

File: README.md (hunk 7 above)
Comment (karthikbekalp) [reply README.md]: Makes sense, updated.

**Review by crowecawcaw — APPROVED** [review]

**Review by joel-wong-aws — APPROVED** [review]

**Pull request merged**